spec/models/tag_spec.rb

```diff
@@ -0,0 +1,11 @@
+describe Tag do
+  context "callbacks" do
+    %i[create destroy].each do |kind|
+      it "calls update_portfolio_stats on #{kind}" do
+        expect(subject).to receive(:update_portfolio_stats)
```

Comment:
This should change to expect to have each of the related `portfolio`s (which you'll need to create as part of the subject) receive `:update_metadata`. Otherwise, you're just stubbing the object under test and we could change the logic in that method completely and this test will still pass.